File: x/wasmx/keeper/testdata/python/simple_storage.py
import json
from wasmx import storage_store, storage_load

def instantiate(initvalue: str):
    print("simple_storage py instantiate", initvalue)

def main(a):
    print("simple_storage main", a)
    input = json.loads(a)
    if "store" in input:
        return store(*input["store"])
    if "load" in input:
        return load()
    raise ValueError('Invalid function')

def store(a: str):
    # value = a.encode()
    # key = "pystore".encode()
    # storage_store(key, value)
    print("store value", a)
    storage_store("pystore", a)

def load() -> str:
    # key = "pystore".encode()
    # return bytearray.decode(value)
    value = storage_load("pystore")
    print("---load value", value)
    return value

File: x/wasmx/keeper/testdata/tinygo/simple_storage/simple_storage.go
package main

import (
	"github.com/tidwall/gjson"

	wasmx "github.com/wasmx/wasmx-go"
)

//go:wasm-module simplestorage
//export instantiate
func instantiate() {
	data := wasmx.GetCallData()
	key := []byte("storagekey")
	wasmx.StorageStore(key, data)
}

type Calldata struct {
	Store []string `json:"store,omitempty"`
	Load  []string `json:"load,omitempty"`
}

func main() {
	data := string(wasmx.GetCallData())
	if gjson.Get(data, "store").Exists() {
		value := gjson.Get(data, "store|0")
		storageStore([]byte(value.String()))
	} else if gjson.Get(data, "load").Exists() {
		resp := storageLoad()
		wasmx.SetReturnData(resp)
	}
}

func storageStore(value []byte) {
	key := []byte("storagekey")
	wasmx.StorageStore(key, value)
}

func storageLoad() []byte {
	key := []byte("storagekey")
	return wasmx.StorageLoad(key)
}
